utils/telemetry/tracker.go

```diff
@@ -0,0 +1,58 @@
+package telemetry
+
+import (
+	"fmt"
+	"github.com/mitchellh/cli"
+	"gopkg.in/segmentio/analytics-go.v3"
+)
+
+var (
+	segmentWriteKey string //value will be injected at build-time
+)
+
+// REALMC-7243 same as the new cli Segment tracking
+type Tracker interface {
+	Track(event event)
+	Close() error
+}
+
+type noopTracker struct{}
+
+func (n *noopTracker) Track(event event) {}
+func (n *noopTracker) Close() error      { return nil }
+
+type segmentTracker struct {
+	client analytics.Client
+	ui     cli.Ui
+}
+
+func newSegmentTracker(ui cli.Ui) Tracker {
+	if len(segmentWriteKey) == 0 {
+		return &noopTracker{}
+	}
+	client := analytics.New(segmentWriteKey)
+	return &segmentTracker{client, ui}
+}
+
+func (s *segmentTracker) Track(event event) {
+	properties := make(map[string]interface{}, len(event.data)+2)
+	properties[eventDataKeyCommand] = event.command
+	properties[eventDataKeyExecutionID] = event.executionID
+
+	for _, datum := range event.data {
+		properties[datum.Key] = datum.Value
+	}
+	if err := s.client.Enqueue(analytics.Track{
+		MessageId:  event.id,
+		Timestamp:  event.time,
+		Event:      string(event.eventType),
+		UserId:     event.userID,
+		Properties: properties,
+	}); err != nil {
+		s.ui.Info(fmt.Sprintf("failed to send Segment event %q: %s", event.eventType, err))
```

Comment:
we should change this from `Segment` to `tracking`

Reply:
this may even just want to fail silently?  Chris was talking to me about this while I was OOO in regards to the new CLI, but some of these errors/warnings can potentially be noisy and unnecessary to the end user.  they cant do anything about it (if its failing)

Reply:
I'm fine with failing silently too

Reply:
yeah, IMO this is one of the few cases where failing silently is the right idea. we really don't want to be littering CLI output with these failure messages in the event of a non critical failure, especially one that essentially says "Hey! Normally we'd be tracking you, but something is wrong! You can't do anything about it!"

we would know that something is up on our end when we notice we aren't seeing any `1.3.x` events coming in.